src/Examine.Lucene/Suggest/LuceneSuggester.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Lucene.Net.Index;
+
+namespace Examine.Lucene.Suggest
+{
+    /// <summary>
+    /// Lucene Index based Suggester
+    /// </summary>
+    public class LuceneSuggester : BaseLuceneSuggester, IDisposable
+    {
+        private readonly ReaderManager _readerManager;
+        private readonly FieldValueTypeCollection _fieldValueTypeCollection;
+        private readonly SuggesterDefinitionCollection _suggesterDefinitions;
+        private bool _disposedValue;
+
+        private Dictionary<string, ILookupExecutor> _suggesters = new Dictionary<string, ILookupExecutor>();
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="name">Name of the Suggester</param>
+        /// <param name="readerManager">Retrieves a IndexReaderReference for the index the Suggester is for</param>
+        /// <param name="fieldValueTypeCollection">Index Field Types</param>
+        public LuceneSuggester(string name, ReaderManager readerManager, FieldValueTypeCollection fieldValueTypeCollection, SuggesterDefinitionCollection suggesterDefinitions)
+            : base(name)
+        {
+            _readerManager = readerManager;
+            _fieldValueTypeCollection = fieldValueTypeCollection;
+            _suggesterDefinitions = suggesterDefinitions;
+            BuildSuggesters();
+        }
+
+        public void RebuildSuggesters() => BuildSuggesters(true);
```

Comment:
This is public and should therefore have XML documentation - Use `/// <inheritdoc />` where appropriate

Reply:
Done